### PR TITLE
Apply clang-format to all code

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
@mcfadden8 @kab163 - take a look and comment if there's anything formatted in a way you dislike.

File: src/umpire/Allocator.cpp [hunk 1]
```diff
@@ -6,91 +6,80 @@
 //////////////////////////////////////////////////////////////////////////////
 #include "umpire/Allocator.hpp"
 
+#include "umpire/Replay.hpp"
```
Comment (mcfadden8) [inline src/umpire/Allocator.cpp:9]: Sorting header files too!  Nice.

File: src/umpire/Allocator.cpp [hunk 2]
```diff
@@ -6,91 +6,80 @@
...
+                << m_allocator << "\" }");
 
   UMPIRE_LOG(Debug, "");
 
   m_allocator->release();
 }
 
-std::size_t
-Allocator::getSize(void* ptr) const
+std::size_t Allocator::getSize(void* ptr) const
```
Comment (mcfadden8) [inline src/umpire/Allocator.cpp:35]: I actually prefer this style of keeping the method on the same line as the return type when possible.

Comment (davidbeckingsale) [reply src/umpire/Allocator.cpp:35]: Haha this is the one thing I don't really like 😄

File: src/umpire/Replay.cpp [hunk 3]
```diff
@@ -5,50 +5,45 @@
...
-  replayEnabled =  enable_replay;
+  replayEnabled = enable_replay;
 }
 
-void Replay::logMessage( const std::string& message )
+void Replay::logMessage(const std::string& message)
 {
-  if ( !replayEnabled )
-    return;   /* short-circuit */
+  if (!replayEnabled) return; /* short-circuit */
```
Comment (mcfadden8) [inline src/umpire/Replay.cpp:41]: hmm.  this is the only one I see that may be problematic @davidbeckingsale.  I would prefer:
```suggestion
  if (!replayEnabled) {
    return; /* short-circuit */
  }
```

File: src/umpire/Replay.cpp [hunk 4]
```diff
@@ -5,50 +5,45 @@
...
 
   umpire::replay() << message;
 }
 
-bool Replay::replayLoggingEnabled()
-{
-  return replayEnabled;
-}
+bool Replay::replayLoggingEnabled() { return replayEnabled; }
```
Comment (mcfadden8) [inline src/umpire/Replay.cpp:46]: Same comment as above.  I would prefer:
```suggestion
bool Replay::replayLoggingEnabled()
{
  return replayEnabled;
}
```

**Comment by davidbeckingsale** [comment]
@mcfadden8 - I think this addresses your comments.

File: src/umpire/Replay.cpp (hunk 3 above)
Comment (mcfadden8) [reply src/umpire/Replay.cpp]: @davidbeckingsale, this one still seems to be present.  Can we modify it to add curly braces around open if statements?

File: src/umpire/DeviceAllocator.cpp [hunk 5]
```diff
@@ -12,33 +12,31 @@
...
   auto device_alloc = rm.getAllocator(umpire::resource::Device);
 
-  m_counter = static_cast<unsigned int*>(
-    device_alloc.allocate(sizeof(unsigned int)));
+  m_counter =
+      static_cast<unsigned int*>(device_alloc.allocate(sizeof(unsigned int)));
   rm.memset(m_counter, 0);
 }
 
 __host__ __device__
```
Comment (kab163) [inline src/umpire/DeviceAllocator.cpp:29]: Line 29 has `__host__ __device` on it's own line while Line 39 had this changed. Should this be changed?

Comment (davidbeckingsale) [reply src/umpire/DeviceAllocator.cpp:29]: I think it remains split due to line length.

Comment (kab163) [reply src/umpire/DeviceAllocator.cpp:29]: Yea, that seems likely. If you see DeviceAllocator.hpp, line 30, I think that looks a lot better. But this is ok :p

**Comment by kab163** [comment]
So I wanted to double check: if statements, classes, and for loops have 
`if (something) {
...
}`
but functions have
`int main () 
{
...
}`
that seems to be the trend.

**Comment by davidbeckingsale** [comment]
LGTM

File: src/umpire/Replay.cpp (hunk 3 above)
Comment (mcfadden8) [reply src/umpire/Replay.cpp]: Resolving this since the reformatting tool only makes white-space changes by design.

**Comment by davidbeckingsale** [comment]
LGTM

**Pull request merged**